adapters/dysonfinance/src/index.ts

```diff
@@ -0,0 +1,258 @@
+	const hexBlockNumber = "0x" + number.toString(16); // Convert decimal block number to hexadecimal
+	console.log(hexBlockNumber);
+	const data = await post(LINEA_RPC, {
+		jsonrpc: "2.0",
+		method: "eth_getBlockByNumber",
+		params: [hexBlockNumber, false],
+		id: 1,
+	});
+	const blockTimestampInt = parseInt(data.result.timestamp);
+	return blockTimestampInt;
+};
+
+const getPoolsData = async (
+	blockNumber: number,
+	blockTimestamp: number,
+): Promise<Map<string, any>> => {
+	const queryBlock = blockNumber > 675342 ? blockNumber : 675342;
+	const responseJson = await post(DYSON_SUBGRAPH_QUERY_URL, {
+		query: DYSON_POOLS_QUERY,
+		variables: { blockNumber: queryBlock },
+	});
+	const poolMap = new Map<string, any>();
+
+	const poolArray = responseJson?.data?.pairEntities || [];
+
+	for (let index = 0; index < poolArray.length; index++) {
+		const element = poolArray[index];
+		poolMap.set(element?.id, element);
+	}
+	return poolMap;
+};
+
+function formatUnits(value: bigint, decimals: number) {
+    let display = value.toString()
+
+    const negative = display.startsWith('-')
+    if (negative) display = display.slice(1)
+
+    display = display.padStart(decimals, '0')
+
+    let [integer, fraction] = [
+      display.slice(0, display.length - decimals),
+      display.slice(display.length - decimals),
+    ]
+    fraction = fraction.replace(/(0+)$/, '')
+    return `${negative ? '-' : ''}${integer || '0'}${
+      fraction ? `.${fraction}` : ''
+    }`
+  }
+
+interface SumPosition {
+	user: string;
+	tokenAddress: string;
+	tokenBalance: number;
+	tokenSymbol: string;
+}
+
+const getPositionData = async (
+	blockNumber: number,
+	blockTimestamp: number,
+): Promise<OutputDataSchemaRow[]> => {
+	const queryBlock = blockNumber > 675341 ? blockNumber : 675341;
+	const poolMap = await getPoolsData(blockNumber, blockTimestamp);
+	const responseJson = await post(DYSON_SUBGRAPH_QUERY_URL, {
+		query: DYSON_POSITIONS_QUERY,
+		variables: { blockNumber: queryBlock },
+	});
+	const userPositionMap = new Map<string, SumPosition>();
+	const positionsArray = responseJson?.data?.noteEntities || [];
+
+	for (let index = 0; index < positionsArray.length; index++) {
+		const element = positionsArray[index];
+		const pairData = poolMap.get(element.pair.toLowerCase());
+		const depositToken =
+			element.depositType === 0
+				? {
+						tokenAddress: pairData.token0Address,
+						tokenSymbol: pairData.token0Name,
+                        decimals: parseFloat(pairData.token0Decimals)
+				  }
+				: {
+						tokenAddress: pairData.token1Address,
+						tokenSymbol: pairData.token1Name,
+                        decimals: parseFloat(pairData.token1Decimals)
+				  };
+		const id = `${element.user}-${depositToken.tokenAddress}`
+        let position: SumPosition | undefined = userPositionMap.get(id)
+        const depositAmt = parseFloat(formatUnits(element.depositAmt, depositToken.decimals))
```

Comment:
are these currently formatting the numbers to real numbers?

we would want raw numbers here

Reply:
@ytwang00

Reply:
Yes, it's formatted to real numbers.
I'll fix it very soon.